### ISSUE TITLE
[spend allocation] Clarify various points with more/expanded examples

### ISSUE THREAD

**Issue by lobsterkatie** [getsentry/sentry-docs#8663]
### Core or SDK?

Core Sentry product

### Which part? Which one?

Spend allocation

### Description

On the [Spend Allocation page](https://docs.sentry.io/product/accounts/quotas/spend-allocation/), there are a few places where I think we could be clearer about things.

1. We give an example of spend allocation:

    > For example, if your total reserved volume is 1M events, and you have two important projects, you can decide to allocate 300K events to project 1, 250k events to project 2, and leave the remaining unallocated pool of 450k events for all other projects.

    I think this would benefit from a second sentence along the lines of, "This means that if the other projects receive 500K events, the extra 50K won't cut into the the 300K allocated to project 1, but instead either be supported by your on-demand budget or dropped as over quota." (I'm not just adding this sentence myself because I _think_ it's correct, but am unfamiliar with the feature and so wouldn't swear that I'm right.)

    It also might be useful to have an example of the other use case mentioned, that of protecting a small project.

2. (Kind of related) There's a note which says:

    > With Spend Allocation, excess event consumption may come from your on-demand budget, **even if you still have reserved volume available**.

    Again I'm not totally sure, but what I _think_ this means is that, as in my proposed sentence above, events allocated to one project won't be used to handle excess volume in another project. Is that true? Also, I'm not sure whether you're allowed to dip into unallocated events to support overages in a project with allocated events. I'm guessing so, since it says "The per-project allocation you set is the **minimum** not the **maximum** number of events each project will get," but again, it's not totally clear. Maybe two more extra sentences on the given example could clarify this, one each to illustrate the questions I've raised in this paragraph? 

3. It's not 100% clear what would happen if you don't leave any unallocated events, both in the case where you have other projects and in the case where you don't.

In short, there are a number of potential scenarios which could benefit from examples explaining how they'd unfold.

### Suggested Solution

I'm going to delete this section, because the answer is above.
